### PR TITLE
Remove inlines from DepNode code

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
@bors try @rust-timer queue

Let's also get perf stats to see if this is a compile time regression (though I would be surprised).

**Comment by michaelwoerister** [comment]
I'd like to see what happens if we just remove `#[inline(always)]` annotation from `DepNode::new()`. I'm not a big fan of duplicating the code like that.

**Comment by Mark-Simulacrum** [comment]
@bors try @rust-timer queue

Let's get this kicked off as well...

I've tried blanket removing inline from the dep_node module.

**Comment by michaelwoerister** [comment]
The comparison doesn't show anything for me.

I think we should also mark the following functions as `const` (as suggested on Zulip):
- can_reconstruct_query_key
- is_anon
- is_eval_always
- has_params
- from_def_path_hash (no sure if debug_assert is a problem for const-fn)
- new_no_params (ditto)

**Comment by Mark-Simulacrum** [comment]
Compare pages are a bit slower to render, I usually need to wait 1-2 seconds I think these days. It's on my todo list :)

I'll investigate const fn for those; I sort of expect it won't be possible -- I thought `match`, like `if`, is not yet supported in const fn.

**Comment by Mark-Simulacrum** [comment]
I think I confirmed all of these can't be const fn due to insufficient work on const fn; miri probably can evaluate these but we haven't pushed that support into rustc const eval proper just yet.

File: src/librustc/dep_graph/dep_node.rs [hunk 1]
```diff
@@ -219,15 +216,17 @@ macro_rules! define_dep_nodes {
                                     hash
                                 };
 
-                                if cfg!(debug_assertions) &&
-                                   !dep_node.kind.can_reconstruct_query_key() &&
+                                #[cfg(debug_assertions)]
+                                {
+                                if !dep_node.kind.can_reconstruct_query_key() &&
```
Comment (michaelwoerister) [inline src/librustc/dep_graph/dep_node.rs]: Indent please.

File: src/librustc/dep_graph/dep_node.rs [hunk 2]
```diff
@@ -242,15 +241,17 @@ macro_rules! define_dep_nodes {
...
```
Comment (michaelwoerister) [inline src/librustc/dep_graph/dep_node.rs]: This too.

**Review by michaelwoerister — APPROVED** [review]
> I think I confirmed all of these can't be const fn due to insufficient work on const fn; miri probably can evaluate these but we haven't pushed that support into rustc const eval proper just yet.

That's a pity because almost all invocations have constant parameters. In the future maybe.

Thanks for the PR, @Mark-Simulacrum! r=me with the indentation fixed.

**Comment by Mark-Simulacrum** [comment]
@bors r=michaelwoerister

Yeah, it is a bit unfortunate.

**Pull request merged**